### PR TITLE
Gradient updates with Alea

### LINKED ISSUE #383: Making `gradient` more robust
Gradient's docstring says it accepts numpy arrays or xr.Datarrays, but it only accepts Xarrays with 1D lat lon coordinates. We want to make the inputs more robust so the user can pass in a numpy array with lat and lon kwargs, or the user can pass in an Xarray object where 1D or 2D lat lon coordinates is inferred or overwritten with kwargs.

**To Reproduce**
Passing in a numpy array current yields
`AttributeError: 'numpy.ndarray' object has no attribute 'coords'`

Passing in an Xarray with 2D coords yields
`ValueError: conflicting sizes for dimension 'x': length 80 on the data but length 8 on coordinate 'lon'`
** except 8 and 80 are derived from the lon shape (here (8,10))

**Steps**
@pilotchute and I are going to pair program this on Monday morning to add numpy, kwargs, and dimension checks.

### PR DESCRIPTION
**Pull request by jukent**
## PR Summary
Closes #383 

## PR Checklist

**General**
- [x] Make an issue if one doesn't already exist
- [x] Link the issue this PR resolves by adding `closes #XXX` to the PR description where XXX is the number of the issue.
- [x] Add appropriate labels to this PR
- [x] Make your changes in a forked repository rather than directly in this repo
- [x] Open this PR as a draft if it is not ready for review
- [x] Convert this PR from a draft to a full PR before requesting reviewers
- [x] Request `@NCAR/geocat` for reviews
- [x] Passes `precommit`. To set up on your local, run `pre-commit install` from the top level of the repository. To manually run pre-commits, use `pre-commit run --all-files` and re-add any changed files before committing again and pushing.

**Functionality**
- [x] Function is in appropriate module file
- [x] New function(s) intended for public API added to `src/geocat/comp/__init__.py` file

**Testing**
- [x] Tests for function exists in associated module test file
- [x] Tests cover all possible logical paths in your function

**Documentation**
- [x] Docstrings have been added to all new functions ([Documentation Standards](https://geocat.ucar.edu/pages/contributing.html#422-documentation))
- [x] Docstrings have updated with any function changes
- [x] Internal functions have a preceding underscore (`_`) and have been added to `docs/internal_api/index.rst`
- [x] User facing functions have been added to `docs/user_api/index.rst` under their module

### CODE REVIEW

File: test/test_gradient.py [hunk 1]
```diff
@@ -28,48 +35,164 @@ class Test_Gradient(unittest.TestCase):
...
+        )
+
+    def test_gradient_axis1_np_2d_nocoords(self):
+        self.results = gradient(self.test_data_np, self.test_coords_2d_lon_np,
+                                self.test_coords_2d_lat_np)
         self.results_axis1 = self.results[1]
         np.testing.assert_almost_equal(
-            self.results_axis1.data,
-            self.test_results_lat.data,
+            self.results_axis1,
```
Comment (pilotchute) [inline test/test_gradient.py:195]: this line should have `.values`

Comment (jukent) [reply test/test_gradient.py:195]: Here self.results_axis1 is a numpy array, so `.values` causes it to fail.

Comment (pilotchute) [reply test/test_gradient.py:195]: oh right. 
thanks

**Comment by anissa111** [comment]
I know this PR was started before the most recent PR template change, but don't forget to add a summary of these changes to `docs/release-notes.rst`

**Review by anissa111 — APPROVED** [review]

**Pull request merged**